node-runner/src/lib.rs

```diff
@@ -166,7 +166,15 @@ pub fn init_program<E: Environment<Ext>>(
         value,
         reply: None,
     };
-    let result = RunNextResult::from_single(init_message, run_result);
+    let mut result = RunNextResult::from_single(init_message, run_result.clone());
+
+    result.messages.append(
+        &mut run_result
+            .messages
+            .into_iter()
+            .map(|msg| msg.into_message(program_id))
+            .collect(),
```

Comment:
Looks pretty good, but I would suggest to refactor `Run(Next)Restut` & `from_single`. `RunResult` sould contain the processed message and `from_single` should have just one argument - `RunResult`. `messages` have to be populated withing `from_single`.

I believe this code is duplicated already =)

Reply:
All this things will be refactored by @breathx